### DISCUSSION TITLE
Connection with TickTick through API token

### DISCUSSION THREAD

**Discussion by fducros** [thesamim/TickTickSync#100] [Ideas]
Hi there, your plugin looks fantastic. But for a host of reasons, it is both scary (security?) and impractical (I login through google auth0) to put a login name and password. when do you plan to enable the integration through API token? :)

Thank you thank you :)

**Comment by thesamim** [comment]
Pardon the delayed response.

> Hi there, your plugin looks fantastic. 

Thanks!

> But for a host of reasons, it is both scary (security?) and impractical (I login through google auth0) to put a login name and password. when do you plan to enable the integration through API token? :)
> 

TickTick's API token enabled API is V1.

TickTickSync functionality is supported by their V2 API.

If TickTick ever enables API Token access to their V2 API, I will look into the conversion at that point.

For what it's worth: I understand the concern. But that UserId and Password are not stored anywhere. TickTickSync caches an authentication token received during login. If that expires, you will have to login again, because the UserId and Password are not stored.

> Thank you thank you :)

Welcome!

**Reply by fducros** [reply]
OK. Being a premium user, I am asking them. Thank you.
